src/ip/QosConfig.cc

```diff
@@ -346,14 +345,12 @@ Ip::Qos::Config::parseConfigLine()
 
             if (mark) {
                 if (!xstrtoui(&token[10], nullptr, &markLocalHit, 0, std::numeric_limits<nfmark_t>::max())) {
-                    debugs(3, DBG_CRITICAL, "ERROR: Bad mark local-hit value " << &token[10]);
-                    self_destruct();
+                    throw TextException(ToSBuf("Bad mark local-hit value ", &token[10]), Here());
                 }
             } else {
                 unsigned int v = 0;
                 if (!xstrtoui(&token[10], nullptr, &v, 0, std::numeric_limits<tos_t>::max())) {
-                    debugs(3, DBG_CRITICAL, "ERROR: Bad TOS local-hit value " << &token[10]);
-                    self_destruct();
+                    throw TextException(ToSBuf("Bad TOS local-hit value ", &token[10]), Here());
```

Comment:
The `&token[10]` expression is ugly and dangerous, but I do not think this PR should change that because a proper fix will require some out-of-scope refactoring. Most of these lines will need to be changed again as we migrate to better parsing routines anyway (and that migration will remove that expression), so there is no particular pressure to improve them further right now, in this simple PR IMO. I am not requesting any related changes here.